src/services/nexboard/utils/Nexboard.js

```diff
@@ -0,0 +1,123 @@
+const rp = require('request-promise-native');
+const { BadRequest } = require('@feathersjs/errors');
+const logger = require('../../../logger');
+
+// TODO: use config
+const {
+	NEXBOARD_URL = 'https://nexboard.nexenio.com/',
+	NEXBOARD_URI = 'portal/api/v1/public/',
+	NEXBOARD_API_KEY,
+	NEXBOARD_USER_ID,
+} = process.env;
+
+class Nexboard {
+	constructor({
+		apiKey = NEXBOARD_API_KEY,
+		userID = NEXBOARD_USER_ID,
+		url = NEXBOARD_URL,
+		uri = NEXBOARD_URI,
+	} = {}) {
+		if (!apiKey || !userID || !url || !uri) {
+			logger.error('Missing nextboard envs:', {
+				apiKey, userID, url, uri,
+			});
+		}
+
+		this.apiKey = apiKey;
+		this.user = userID;
+		this.url = url + uri;
+
+		logger.info('Nextboard is set to=', this.url);
+
+		this.err = {
+			projectIds: 'Could not retrieve ProjectIds',
+			createProject: 'Could not create new Project',
+			retrieveBoardsFromProject: 'Could not retrieve Boards from Project',
+			retrieveBoards: 'Could not retrieve Board',
+			createBoard: 'Could not create a new Board',
+		};
+	}
+
+	createSettings({
+		method = 'GET',
+		endpoint,
+		qs = {
+			token: this.apiKey,
+			userId: this.user,
+		},
+		body,
+	}) {
+		return {
+			method,
+			uri: `${this.url}${endpoint}`,
+			qs,
+			body,
+			json: true,
+		};
+	}
+
+	getProject(projectId) {
+		return rp(this.createSettings({ endpoint: `projects/${projectId}` }));
```

Comment:
any handling necessary in error case?

Reply:
It should throw an error to the services that is handle bei feathers.